core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -120,12 +120,14 @@ void HIP::impl_finalize() {
 
   desul::Impl::finalize_lock_arrays();  // FIXME
 
-  for (const auto hip_device : Impl::HIPInternal::hip_devices) {
-    KOKKOS_IMPL_HIP_SAFE_CALL(hipSetDevice(hip_device));
-    KOKKOS_IMPL_HIP_SAFE_CALL(
-        hipEventDestroy(Impl::HIPInternal::constantMemReusable[hip_device]));
-    KOKKOS_IMPL_HIP_SAFE_CALL(
-        hipHostFree(Impl::HIPInternal::constantMemHostStaging[hip_device]));
+  // TODO C++20 Use std::views::values.
+  for (const auto [_, ptr] : Impl::HIPInternal::constantMemHostStaging) {
+    KOKKOS_IMPL_HIP_SAFE_CALL(hipHostFree(ptr));
+  }
+
+  // TODO C++20 Use std::views::values.
+  for(auto& [_, lock] : Impl::HIPInternal::constantMemReusable) {
```

Comment:
```suggestion
  for (auto& [_, lock] : Impl::HIPInternal::constantMemReusable) {
```
(just to be consistent with the style just above)